### ISSUE TITLE
Field order in TypeScript `type` is reversed in generated OpenAPI spec.

### ISSUE THREAD

**Issue by cakoose** [lukeautry/tsoa#820]
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

When I define a TypeScript type with `interface`, the generated OpenAPI definition has the correct field order.  When I define it with `type`, the OpenAPI definition has the fields in reverse order.

```typescript
type Blah1 = {
    a: number,
    b: string,
};

interface Blah2 {
    a: number,
    b: string,
}
```

In "swagger.json":
```json
"Blah2": {
	"properties": {
		"a": {
			"type": "number",
			"format": "double"
		},
		"b": {
			"type": "string"
		}
	},
	"required": [
		"a",
		"b"
	],
	"type": "object",
	"additionalProperties": false
},
"Blah1": {
	"properties": {
		"b": {
			"type": "string"
		},
		"a": {
			"type": "number",
			"format": "double"
		}
	},
	"required": [
		"b",
		"a"
	],
	"type": "object"
},
```

Not sure if the TypeScript API exposes a stable field order...

## Current Behavior

Included above.

## Possible Solution

Reverse the field order.  Add a test to make sure the field order exposed by TypeScript doesn't change unexpectedly.  (Though not sure how to make sure it's tested against all TypeScript versions...)

## Steps to Reproduce

Please let me know if the "Expected Behavior" section isn't enough information to reproduce.

## Context (Environment)

I'm using Yarn and have installed:
- typescript 4.0.3
- tsoa 3.4.0, which seems to resolve to typescript 3.9.7

NodeJS 14.13.1.

- Confirm you were using yarn not npm: [x]

## Detailed Description

Included in "Possible Solution" section.

## Breaking change?

Maybe?  For example, people might work around this issue by reversing the field order TypeScript to get the correct order in the OpenAPI spec.

**Comment by WoH** [comment]
Why would this be a bug?

**Comment by cakoose** [comment]
Are you asking why this is a problem?  Or why I filed this as a bug and not a feature request?

*Why it's a problem:* When I define my TypeScript types, I define the fields in an order that I believe is intuitive, e.g. more important field first.  I'd like that field order to be preserved in the OpenAPI spec so that the field order in the generated documentation and generated client code is also intuitive.

*Why I filed this as a bug and not a feature request:* I think most people would consider the field order in docs to be important, and that the intent was always to preserve the field order.  Since the field order is preserved for `interface` and the tsoa docs generally use `interface` and not `type`, I assumed the reversed field order was an oversight.
